### ISSUE TITLE
Issues from code review 🐛

### ISSUE THREAD

**Issue by KarelZe** [KarelZe/thesis#116]
- Removes outdated notebook
- Unifies some cells
- Fixes import sorting
- Removes Google Colab badges
- Completes doc strings
- Unified use of `|` instead of `Union[...]` or `Optional[...]`
- Replaces some lengthy code with list comprehensions

Adressed in other issues:
- try out moving window approach. See #97. 
- try out learning rate scheduling. See #7.
- change progress bar. See #97.
- update variable name of logits. See #118 .
